hera_qm/firstcal_metrics.py

```diff
@@ -0,0 +1,579 @@
+            if plot_type == 'both':
+                axes = ax
+                ax = axes[0]
+            plabel = []
+            ax.grid(True)
+            for i, index in enumerate(plot_ants):
+                p, = ax.plot(self.frac_JD, self.delays[index], marker='.',
+                             c=cmap[i], **plt_kwargs)
+                plabel.append(p)
+            ax.set_xlabel('fraction of JD %d' % self.start_JD, fontsize=14)
+            ax.set_ylabel('delay solution [ns]', fontsize=14)
+            if plot_type == 'both':
+                ax = axes
+
+        # plot delay offset
+        if (plot_type == 'both') or (plot_type == 'offset'):
+            if plot_type == 'both':
+                axes = ax
+                ax = axes[1]
+            plabel = []
+            ax.grid(True)
+            for i, index in enumerate(plot_ants):
+                p, = ax.plot(self.frac_JD, self.delay_offsets[index],
+                             marker='.', c=cmap[i], **plt_kwargs)
+                plabel.append(p)
+            ax.set_xlabel('fraction of JD %d' % self.start_JD, fontsize=14)
+            ax.set_ylabel('delay offset [ns]', fontsize=14)
+            if plot_type == 'both':
+                ax = axes
+
+        # add legend
+        if custom_ax is False:
+            ax = fig.add_axes([1.0, 0.1, 0.05, 0.8])
+            ax.axis('off')
+            ax.legend(plabel, [self.ants[i] for i in plot_ants])
+        else:
+            ax[-1].legend(plabel, [self.ants[i] for i in plot_ants])
+
+        if save is True and custom_ax is False:
+            if fname is None:
+                fname = self.file_stem + '.dlys.png'
+            fig.savefig(fname, bbox_inches='tight')
+
+    def plot_zscores(self, fname=None, plot_type='full', cm='viridis_r', ax=None, figsize=(10, 6),
+                     save=False, kwargs={'cmap': 'viridis_r', 'vmin': 0, 'vmax': 5}):
+        """
+        Plot absolute value of antenna delay solution z_scores
+
+        Input:
+        ------
+
+        fname : str, default=None
+            filename
+
+        plot_type : str, default='full'
+            Type of plot to make
+            'full' : plot zscore for each (N_ant, N_times)
+            'time_avg' : plot zscore for each (N_ant,) avg over time
+
+        cm : str, default='viridis_r'
+            colormap
+
+        ax : axis object, default=None
+            matplotlib axis object
+
+        figsize : tuple, default=(10,6)
+            figsize if creating figure
+
+        save : bool, default=False
+            save figure to file
+
+        kwargs : dict
+            plotting kwargs
+        """
+
+        custom_ax = True
+        if ax is None:
+            custom_ax = False
+            fig = plt.figure(figsize=figsize)
+            ax = fig.add_subplot(111)
+
+        # Get zscores
+        ant_avg, ant_std, time_std, agg_std, z_scores = self.delay_std()
+        z_scores = np.abs(z_scores)
```

Comment:
Now that you're saving the sign information of the z-scores (which is great!), maybe you could have plotting the absolute value be an option of the routine (perhaps the default), and an option to plot the signed z-score. Just make sure that in the latter case, you switch to a diverging color map (like `spectral` in your previous routine) instead of `viridis_r`. (Side note: in my book, you get +10 points for using `viridis` instead of `jet` ;) )